llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -1337,8 +1337,6 @@ NVPTXTargetLowering::getPreferredVectorAction(MVT VT) const {
   if (!VT.isScalableVector() && VT.getVectorNumElements() != 1 &&
       VT.getScalarType() == MVT::i1)
     return TypeSplitVector;
-  if (Isv2x16VT(VT))
-    return TypeLegal;
   return TargetLoweringBase::getPreferredVectorAction(VT);
```

Comment:
AFAICT, `TargetLoweringBase::getPreferredVectorAction(VT)` will return TypePromoteInteger, which is not what we want here.
https://github.com/llvm/llvm-project/blob/a18af41c20ac9ca22e3c95da3d71475f9f6c31b5/llvm/include/llvm/CodeGen/TargetLowering.h#L518

Reply:
Yes, that's a good point. 

My description should have been more accurate. The reason we should delete this isn't because we should rely on the default implementation of `getPreferredVectorAction()`; it's because it's dead code. 

[Before calling `getPreferredVectorAction()`](https://github.com/llvm/llvm-project/blob/a78861fc55d18046989ff4d624a037e9181da170/llvm/lib/CodeGen/TargetLoweringBase.cpp#L1406), we first [check whether the type is legal](https://github.com/llvm/llvm-project/blob/a78861fc55d18046989ff4d624a037e9181da170/llvm/lib/CodeGen/TargetLoweringBase.cpp#L1399-L1400). Thus, 
```
if (Isv2x16VT(VT))
    return TypeLegal;
```
will never be triggered.

Reply:
OK.